File: barkr/models/message_metadata.py
"""
Module to implement a generic, wrapper model for message metadata.

Message metadata contains optional information about a message
such as language, visibility, and allowed replies. This data
is separated from the core message to keep the Message class
focused on the essential message content.
"""

from dataclasses import dataclass
from typing import Optional

from barkr.models.message_allowed_replies import MessageAllowedReplies
from barkr.models.message_visibility import MessageVisibility


@dataclass(frozen=True)
class MessageMetadata:
    """
    A container for optional message metadata such as
    language, label, visibility, and allowed replies.
    """

    language: Optional[str] = None
    label: Optional[str] = None
    visibility: MessageVisibility = MessageVisibility.PUBLIC
    allowed_replies: Optional[list[MessageAllowedReplies]] = None

File: models/message_metadata.go
package models

// MessageMetadata represents the optional metadata of a [Message] based on its source
// [Connection], including its language, label, visibility, allowed replies, and mentions.
//
// This metadata can be used to enrich the message while it's being published on a target
// platform, and each [Connection] implementation can decide how to map this metadata, if
// applicable, to the target platform's [Message] properties.
//
// The zero value represents a public [Message] with no specific language, label, or allowed replies, and no mentions.
type MessageMetadata struct {
	Language       string                // language of the [Message]
	Label          string                // content label in use for the [Message]
	Visibility     MessageVisibility     // visibility of the [Message] (who can see it) in the source platform
	AllowedReplies MessageAllowedReplies // who can reply to the [Message] in the source platform
	Mentions       []MessageMention      // users mentioned in the [Message]
}
